### ISSUE TITLE
Add a Competitive Programming Algorithm.

### ISSUE THREAD

**Issue by ShivamSinghal1** [ShivamSinghal1/Competitive-Programming-Algorithms#10]
- Add a Competitive Programming Algorithm other than the mentioned in the repository. 
- Please do not copy from the internet and then paste it here. 
- Please try to implement on your own. 
- Try to generalize the Algorithm.
- You will surely going to learn something from that.

**Comment by herobank110** [comment]
Hello, can I try add merge sort?

**Comment by ShivamSinghal1** [comment]
> Hello, can I try add merge sort?

Yes you can , add this in Sorting Folder, also implement it on your own, not just copy-paste.

**Comment by hritik5102** [comment]
Hello @ShivamSinghal1 , I would like to contribute to this issue :) 

**I had few query** 

- Should i create a seperate folder or just add the file ?
- is CPP is mandatory language to use ?
- Should I also update the ReadMe ?

**Comment by ShivamSinghal1** [comment]
> * add the file

Hi , 
If possible try to add it in separate folder, best possible folder. 
No CPP is not mandatory but recommended. 
Update the ReadMe too.

The main purpose that we can learn something from this. 
Hope this help and Thanks for your contribution.

**Comment by rasacharjee** [comment]
Hello @ShivamSinghal1 can I add the Level Order Traversal of a tree

**Comment by hritik5102** [comment]
@ShivamSinghal1 Yep sure :) , I'm adding all sorting algorithm in the sorting folder with ReadMe of all algorithm 

can you please create a seperate issue related to documentation so that i can make PR thier , so that chance of ambiguity will be less

Thank you

**Comment by ShivamSinghal1** [comment]
> Hello @ShivamSinghal1 can I add the Level Order Traversal of a tree

No Please , if we see the generalized version of Level Order Traversal. it will be BFS. and BFS is already implemented. Also The main purpose is to include the competitive Programming Algorithm.

Thanks

**Comment by ShivamSinghal1** [comment]
> @ShivamSinghal1 Yep sure :) , I'm adding all sorting algorithm in the sorting folder with ReadMe of all algorithm
> 
> can you please create a seperate issue related to documentation so that i can make PR thier , so that chance of ambiguity will be less
> 
> Thank you

Okay , But be sure to implement these on your own.

**Comment by hritik5102** [comment]
Yes i have already done that ...  during my placements so all of code is well return and with documentation

**Comment by hritik5102** [comment]
@ShivamSinghal1 Check out PR #12  ... Done with neccasary changes :)

If any furthur changes are required please let me know .. Thank you

**Comment by Code-Recursion** [comment]
i would like to add few algorithms

**Comment by hritik5102** [comment]
> @ShivamSinghal1 Check out PR #12 ... Done with necessary changes :)
> 
> If any further changes are required please let me know .. Thank you

Can you please review the code 

Thank you :)

**Comment by ShivamSinghal1** [comment]
> > @ShivamSinghal1 Check out PR #12 ... Done with necessary changes :)
> > If any further changes are required please let me know .. Thank you
> 
> Can you please review the code
> 
> Thank you :)

I found all your algorithms copied from https://github.com/jainaman224/Algo_Ds_Notes/blob/master/Bubble_Sort/Bubble_Sort.cpp here. 

My main motive that we learn something from it by implementing the algorithms. Please avoid this. This time i am closing your PR for this reason. Please avoid this next time. 

Implement only one algorithms, it will not be a problem or but not copy paste.

**Comment by ShivamSinghal1** [comment]
> > @ShivamSinghal1 Check out PR #12 ... Done with necessary changes :)
> > If any further changes are required please let me know .. Thank you
> 
> Can you please review the code
> 
> Thank you :)

I found all your algorithms copied from https://github.com/jainaman224/Algo_Ds_Notes/blob/master/Bubble_Sort/Bubble_Sort.cpp here. 

My main motive that we learn something from it by implementing the algorithms. Please avoid this. This time i am closing your PR for this reason. Please avoid this next time. 

Implement only one algorithms, it will not be a problem or but not copy paste.

**Comment by ShivamSinghal1** [comment]
> i would like to add few algorithms

Sure, please avoid copy-paste , implement on your own. 
Thanks

**Comment by hritik5102** [comment]
> > > @ShivamSinghal1 Check out PR #12 ... Done with necessary changes :)
> > > If any further changes are required please let me know .. Thank you
> > 
> > 
> > Can you please review the code
> > Thank you :)
> 
> I found all your algorithms copied from https://github.com/jainaman224/Algo_Ds_Notes/blob/master/Bubble_Sort/Bubble_Sort.cpp here.
> 
> My main motive that we learn something from it by implementing the algorithms. Please avoid this. This time i am closing your PR for this reason. Please avoid this next time.
> 
> Implement only one algorithm, it will not be a problem or but not copy-paste.

Ok , yeah fair enough , i'll add my own code but it's not documented ...
and another thing its return in C language , it's works right ?

( Sorry as i'm one of the contributor of the repo (Algo_DS_Notes)  so i thought i should refer from that repo , but yeah you are right , the main purpose of the open source contribution is to learn something .. but if we don't follow the purpose then our contribution doesn't matter )

**Thank you**

**Comment by herobank110** [comment]
@ShivamSinghal1 Please check out PR #16 merge sort algorithm

**Comment by hritik5102** [comment]
@shivamjaiswal011 now you can check i update my commit in the same PR #12

**Comment by hritik5102** [comment]
> @ShivamSinghal1 now you can check i update my commit in the same PR #12

**Comment by dhritippaul** [comment]
Hello @ShivamSinghal1 , I would like to add Binary Exponential code {O(log n) approach}.

**Comment by subhadeeppaul** [comment]
Hello @ShivamSinghal1, I will be happy if I can add Euclidean algorithm GCD.

**Comment by ShivamSinghal1** [comment]
> Hello @ShivamSinghal1 , I would like to add Binary Exponential code {O(log n) approach}.

Already Included

**Comment by dhritippaul** [comment]
Hello @ShivamSinghal1 , I would like to add SPIRAL SEARCH IN A 2d array (C++ program). Shall I add it ?

**Comment by ayush523** [comment]
Hi @ShivamSinghal1 - I would like to add an algorithm to count set bits in an integer . The name of the algorithm is Brian Kernighan Algorithm.

**Comment by ShivamSinghal1** [comment]
> Hi @ShivamSinghal1 - I would like to add an algorithm to count set bits in an integer . The name of the algorithm is Brian Kernighan Algorithm.

Fine.

**Comment by ShivamSinghal1** [comment]
> Hello @ShivamSinghal1 , I would like to add SPIRAL SEARCH IN A 2d array (C++ program). Shall I add it ?

Yes, Fine.

**Comment by Raksha25** [comment]
Hi, @ShivamSinghal1 I would like to add a few solutions to competitive problems like moving zeroes to end in O(n) and constant space, to find non-decreasing array and similar to these. These algorithms are my own and not copied from anywhere although I have submitted it on leetcode to pass test cases but not on any other repo.

**Comment by ShivamSinghal1** [comment]
> e my own a

Sorry , That will not work . You can add it here. https://github.com/seemantaggarwal/HACKTOBERFEST .

**Comment by Raksha25** [comment]
> > e my own a
> 
> Sorry , That will not work . You can add it here. https://github.com/seemantaggarwal/HACKTOBERFEST .

Alright, thanks!

**Comment by subhadeeppaul** [comment]
Yes, I have included my CPP file in Maths section.
Can you please review the code?

Thank You

**Comment by rudrakshi99** [comment]
Hi @ShivamSinghal1, I would like to add Dijkstra Algorithm in C++.

**Comment by Rajat947** [comment]
HI @ShivamSinghal1, can I add quick sort with 3 partition?

**Comment by ShivamSinghal1** [comment]
> Hi @ShivamSinghal1, I would like to add Dijkstra Algorithm in C++.

Fine

**Comment by RashakDude** [comment]
Hey, @ShivamSinghal1 I just added an algorithm for dividing two numbers without using division, multiplication and modulo operation, but just using bit manipulation, which might be useful for interviews. Please review #38

**Comment by panktishah62** [comment]
Hey !
https://github.com/seemantaggarwal/HACKTOBERFEST is marked ineligible by Hacktoberfest as it does not align with their core values. 
Also is https://github.com/ShivamSinghal1/Competitive-Programming-Algorithms/issues/10# this repository valid for contributions by hacktoberfest ??
@ShivamSinghal1

**Comment by ShivamSinghal1** [comment]
> Hey !
> HACKTOBERFEST is marked ineligible by Hacktoberfest as it does not align with their core values. I already made a PR there but could be disqualified due to the above reason. Please do not recommend these types of repositories in the community.
> 
> Also is #10 this repository valid for contributions by hacktoberfest ??
> @ShivamSinghal1

Sorry for that, I don't think so this repository will be ineligible as i merged the only PR that is good and students learn something from that. 

So thats why try to add those algorithm which are unique, coded by you only and others can take help from your written algo.
